lib/engine/game/g_1840/game.rb

```diff
@@ -410,6 +448,91 @@ def upgrades_to?(from, to, special = false, selected_company: nil)
         def needed_exits_for_hex(hex)
           CITY_TRACK_EXITS[hex.id]
         end
+
+        def info_train_name(train)
+          names = train.names_to_prices.keys.sort
+          active_variant = (available_trains & train.variants.keys).first
+          return names.join(', ') unless active_variant
+
+          names -= [active_variant]
+          "#{active_variant}, (#{names.join(', ')})"
+        end
+
+        def info_available_train(_first_train, train)
+          !(available_trains & train.variants.keys).empty?
+        end
+
+        def info_train_price(train)
+          name_and_prices = train.names_to_prices.sort_by { |k, _v| k }.to_h
+
+          active_variant = (available_trains & train.variants.keys).first
+          return name_and_prices.values.map { |p| format_currency(p) }.join(', ') unless active_variant
+
+          active_price = name_and_prices[active_variant]
+          name_and_prices.delete(active_variant)
+
+          "#{active_price}, (#{name_and_prices.values.map { |p| format_currency(p) }.join(', ')})"
+        end
+
+        def available_trains
+          index = [@cr_counter - 1, 0].max
+          TRAIN_ORDER[index]
+        end
+
+        def remove_obsolete_trains
+          train_to_remove = DEPOT_CLEARING[@cr_counter - 1]
+          return unless train_to_remove
+
+          @depot.export_all!(train_to_remove)
+        end
+
+        def buy_train(operator, train, price = nil)
+          super
+
+          new_cost = MAINTAINANCE_COST[train.sym]
+          @active_maintainance_cost = new_cost if new_cost['Y1'] &&
+                                      (@active_maintainance_cost['Y1'].nil? ||
+                                         new_cost['Y1'] < @active_maintainance_cost['Y1'])
+        end
+
+        def status_str(corporation)
+          return if corporation.type != :minor
+
+          "Maintenance: #{format_currency(maintenance_costs(corporation))}"
+        end
+
+        def maintenance_costs(corporation)
+          corporation.trains.inject(0) { |sum, train| sum + train_maintenance(train.sym) }
```

Comment:
Why not corporation.trains.sum { |t| train_maintenance(t.sym) } ?

Reply:
thanks. I searched for it, and assumed it does not work with a lambda expression...